### ISSUE TITLE
error when build with docker

### ISSUE THREAD

**Issue by Li-kewei** [mapillary/OpenSfM#715]
Configuring for python 3.8...
-- A library with LAPACK API found.
-- Found AMD library: /usr/lib/x86_64-linux-gnu/libamd.so
-- Found AMD header in: /usr/include/suitesparse
-- Found CAMD library: /usr/lib/x86_64-linux-gnu/libcamd.so
-- Found CAMD header in: /usr/include/suitesparse
-- Found COLAMD library: /usr/lib/x86_64-linux-gnu/libcolamd.so
-- Found COLAMD header in: /usr/include/suitesparse
-- Found CCOLAMD library: /usr/lib/x86_64-linux-gnu/libccolamd.so
-- Found CCOLAMD header in: /usr/include/suitesparse
-- Found CHOLMOD library: /usr/lib/x86_64-linux-gnu/libcholmod.so
-- Found CHOLMOD header in: /usr/include/suitesparse
-- Found CXSPARSE library: /usr/lib/x86_64-linux-gnu/libcxsparse.so
-- Found SuiteSparseQR library: /usr/lib/x86_64-linux-gnu/libspqr.so
-- Found SuiteSparseQR header in: /usr/include/suitesparse
-- Found Intel Thread Building Blocks (TBB) library: /usr/lib/x86_64-linux-gnu/libtbb.so, assuming SuiteSparseQR was compiled with TBB.
-- Found Intel Thread Building Blocks (TBB) Malloc library: /usr/lib/x86_64-linux-gnu/libtbbmalloc.so
-- Found SuiteSparse_config library: /usr/lib/x86_64-linux-gnu/libsuitesparseconfig.so
-- Found SuiteSparse_config header in: /usr/include/suitesparse
-- Adding librt: /usr/lib/x86_64-linux-gnu/librt.so to SuiteSparse_config libraries (required on Linux & Unix [not OSX] if SuiteSparse is compiled with timing).
-- Did not find METIS library (optional SuiteSparse dependency)
-- No preference for use of exported gflags CMake configuration set, and no hints for include/library directories provided. Defaulting to preferring an installed/exported gflags CMake configuration if available.
-- Found installed version of gflags: /usr/lib/x86_64-linux-gnu/cmake/gflags
-- Detected gflags version: 2.2.2
-- Found OpenCV: /usr (found version "4.2.0") 
-- Found OpenCV: /usr (found version "4.2.0") found components: core imgproc calib3d imgcodecs 
CMake Error at CMakeLists.txt:53 (add_subdirectory):
  The source directory

    /source/OpenSfM/opensfm/src/third_party/pybind11

  does not contain a CMakeLists.txt file.

CMake Error at bundle/CMakeLists.txt:39 (pybind11_add_module):
  Unknown CMake command "pybind11_add_module".

-- Configuring incomplete, errors occurred!
See also "/source/OpenSfM/cmake_build/CMakeFiles/CMakeOutput.log".
See also "/source/OpenSfM/cmake_build/CMakeFiles/CMakeError.log".
Traceback (most recent call last):
  File "setup.py", line 36, in <module>
    configure_c_extension()
  File "setup.py", line 27, in configure_c_extension
    subprocess.check_call(cmake_command, cwd='cmake_build')
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '../opensfm/src', '-DPYTHON_EXECUTABLE=/usr/bin/python3']' returned non-zero exit status 1.

**Comment by Li-kewei** [comment]
/source/OpenSfM/cmake_build/CMakeFiles/CMakeError.log ：

Determining if the function sgemm_ exists failed with the following output:
Change Dir: /source/OpenSfM/cmake_build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make cmTC_a352d/fast && /usr/bin/make -f CMakeFiles/cmTC_a352d.dir/build.make CMakeFiles/cmTC_a352d.dir/build
make[1]: Entering directory '/source/OpenSfM/cmake_build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_a352d.dir/CheckFunctionExists.c.o
/usr/bin/cc   -fopenmp -DCHECK_FUNCTION_EXISTS=sgemm_ -fPIE   -o CMakeFiles/cmTC_a352d.dir/CheckFunctionExists.c.o   -c /usr/share/cmake-3.16/Modules/CheckFunctionExists.c
Linking C executable cmTC_a352d
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_a352d.dir/link.txt --verbose=1
/usr/bin/cc  -fopenmp -DCHECK_FUNCTION_EXISTS=sgemm_    -rdynamic CMakeFiles/cmTC_a352d.dir/CheckFunctionExists.c.o  -o cmTC_a352d
/usr/bin/ld: CMakeFiles/cmTC_a352d.dir/CheckFunctionExists.c.o: in function `main':
CheckFunctionExists.c:(.text+0x14): undefined reference to `sgemm_'
collect2: error: ld returned 1 exit status
make[1]: *** [CMakeFiles/cmTC_a352d.dir/build.make:87: cmTC_a352d] Error 1
make[1]: Leaving directory '/source/OpenSfM/cmake_build/CMakeFiles/CMakeTmp'
make: *** [Makefile:121: cmTC_a352d/fast] Error 2



Determining if the function cheev_ exists failed with the following output:
Change Dir: /source/OpenSfM/cmake_build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make cmTC_77323/fast && /usr/bin/make -f CMakeFiles/cmTC_77323.dir/build.make CMakeFiles/cmTC_77323.dir/build
make[1]: Entering directory '/source/OpenSfM/cmake_build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_77323.dir/CheckFunctionExists.c.o
/usr/bin/cc   -fopenmp -DCHECK_FUNCTION_EXISTS=cheev_ -fPIE   -o CMakeFiles/cmTC_77323.dir/CheckFunctionExists.c.o   -c /usr/share/cmake-3.16/Modules/CheckFunctionExists.c
Linking C executable cmTC_77323
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_77323.dir/link.txt --verbose=1
/usr/bin/cc  -fopenmp -DCHECK_FUNCTION_EXISTS=cheev_    -rdynamic CMakeFiles/cmTC_77323.dir/CheckFunctionExists.c.o  -o cmTC_77323  /usr/lib/x86_64-linux-gnu/libblas.so
/usr/bin/ld: CMakeFiles/cmTC_77323.dir/CheckFunctionExists.c.o: in function `main':
CheckFunctionExists.c:(.text+0x14): undefined reference to `cheev_'
collect2: error: ld returned 1 exit status
make[1]: *** [CMakeFiles/cmTC_77323.dir/build.make:88: cmTC_77323] Error 1
make[1]: Leaving directory '/source/OpenSfM/cmake_build/CMakeFiles/CMakeTmp'
make: *** [Makefile:121: cmTC_77323/fast] Error 2

**Comment by Li-kewei** [comment]
I have solved it by using “git clone --recursive https://github.com/mapillary/OpenSfM
”, The page is https://www.opensfm.org/docs/building.html

**Comment by jgrocha** [comment]
The current build is failing, as of March, 13, 3f0ccda83fd9fcdb0900809fe1ade211b80b5fed
Three weeks ago was building file, on February 25, it was building fine. Between these dates something has changed and the docker build is breaking with:

```
/usr/bin/ld: CMakeFiles/bundle_test.dir/test/bundle_data_test.cc.o: in function `BACameraWithPriorFixture_ReturnPriorData_Test::TestBody()':
bundle_data_test.cc:(.text+0x116c): undefined reference to `Camera::GetParametersValues() const'
/usr/bin/ld: CMakeFiles/bundle_test.dir/test/bundle_data_test.cc.o: in function `BACameraWithPriorFixture_ReturnSigmaData_Test::TestBody()':
bundle_data_test.cc:(.text+0x135c): undefined reference to `Camera::GetParametersValues() const'
/usr/bin/ld: CMakeFiles/bundle_test.dir/test/bundle_data_test.cc.o: in function `BACameraWithPriorFixtureBase::BACameraWithPriorFixtureBase() [clone .constprop.0]':
bundle_data_test.cc:(.text+0x231b): undefined reference to `Camera::CreatePerspectiveCamera(double, double, double)'
/usr/bin/ld: bundle_data_test.cc:(.text+0x2337): undefined reference to `Camera::CreatePerspectiveCamera(double, double, double)'
/usr/bin/ld: bundle_data_test.cc:(.text+0x236f): undefined reference to `Camera::GetParametersValues() const'
/usr/bin/ld: CMakeFiles/bundle_test.dir/test/bundle_data_test.cc.o: in function `BACamera::ValueToData(Camera const&, Eigen::Matrix<double, -1, 1, 0, -1, 1>&) const':
bundle_data_test.cc:(.text._ZNK8BACamera11ValueToDataERK6CameraRN5Eigen6MatrixIdLin1ELi1ELi0ELin1ELi1EEE[_ZNK8BACamera11ValueToDataERK6CameraRN5Eigen6MatrixIdLin1ELi1ELi0ELin1ELi1EEE]+0x20): undefined reference to `Camera::GetParametersValues() const'
/usr/bin/ld: CMakeFiles/bundle_test.dir/test/bundle_data_test.cc.o: in function `BACamera::DataToValue(Eigen::Matrix<double, -1, 1, 0, -1, 1> const&, Camera&) const':
bundle_data_test.cc:(.text._ZNK8BACamera11DataToValueERKN5Eigen6MatrixIdLin1ELi1ELi0ELin1ELi1EEER6Camera[_ZNK8BACamera11DataToValueERKN5Eigen6MatrixIdLin1ELi1ELi0ELin1ELi1EEER6Camera]+0x8): undefined reference to `Camera::SetParametersValues(Eigen::Matrix<double, -1, 1, 0, -1, 1> const&)'
/usr/bin/ld: CMakeFiles/bundle_test.dir/test/bundle_data_test.cc.o: in function `BACameraWithPriorFixtureBase::BACameraWithPriorFixtureBase()':
bundle_data_test.cc:(.text._ZN28BACameraWithPriorFixtureBaseC2Ev[_ZN28BACameraWithPriorFixtureBaseC5Ev]+0x4f): undefined reference to `Camera::CreatePerspectiveCamera(double, double, double)'
/usr/bin/ld: bundle_data_test.cc:(.text._ZN28BACameraWithPriorFixtureBaseC2Ev[_ZN28BACameraWithPriorFixtureBaseC5Ev]+0x6b): undefined reference to `Camera::CreatePerspectiveCamera(double, double, double)'
/usr/bin/ld: bundle_data_test.cc:(.text._ZN28BACameraWithPriorFixtureBaseC2Ev[_ZN28BACameraWithPriorFixtureBaseC5Ev]+0xa3): undefined reference to `Camera::GetParametersValues() const'
collect2: error: ld returned 1 exit status
make[2]: *** [bundle/CMakeFiles/bundle_test.dir/build.make:178: bundle/bundle_test] Error 1
make[1]: *** [CMakeFiles/Makefile2:555: bundle/CMakeFiles/bundle_test.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by RuslanOm** [comment]
Have same problem

**Comment by jgrocha** [comment]
The problem was introduced between March 4 and March 8.

My build attempts:

OK

Checking out the repo after `fix: gcp.coordinates never None. Check has_value` (#716) was merged, the **build works** (Thu Mar 4)

```
git checkout 576809216328dd5b7db0eb411cc696df03254e24
docker build -t mapillary/opensfm .
```

NOT OK

Checking out the repo after `Fixed export_geocoords` (#710) was merged, the **build fails** (Mon Mar 8)
```
git checkout 7a8a820e884b0ce914c8cec14c21ed1880bfcfb1
docker build -t mapillary/opensfm .
```

**Comment by YanNoun** [comment]
Hi, 

Build is fixed here : https://github.com/mapillary/OpenSfM/pull/723. Should be in master this monday.

Yann
